### PR TITLE
[DAP] [VECTORIZATION PASS] Iir vectorization pass

### PR DESCRIPTION
**Pull request by taiqzheng**
- Add a vectorization pass for DAP Dialect.
 When a machine support SIMD, this pass can run 10x faster than LowerDAPPass.
- Add DAPUtils to support different vector length. 
 Currently support vector length {4, 8, 16, 32, 64}, all kernel.dim0<=64 will use a suitable vector length in processing.

### CODE REVIEW

**Comment by taiqzheng** [comment]
I will add a README.txt for algorithm explanation soon.

**Comment by taiqzheng** [comment]
> I will add a README.txt for algorithm explanation soon.

Now everthing is okay.

File: midend/include/Utils/DAPUtils.h
Comment (zhanghb97) [inline midend/include/Utils/DAPUtils.h:1]: clang-format

File: midend/lib/Conversion/DAPVectorization/DAPVectorization.cpp [hunk 1]
```diff
@@ -0,0 +1,210 @@
+//====- DAPVectorization.cpp - DAP Dialect Vectorization Pass  ------------===//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (zhanghb97) [inline midend/lib/Conversion/DAPVectorization/DAPVectorization.cpp:2]: Add an empty line here.

File: midend/lib/Conversion/DAPVectorization/DAPVectorization.cpp [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+    Value cond4 =
+        rewriter.create<CmpIOp>(loc, CmpIPredicate::ule, filterSize, c4);
+    Value cond8 =
+        rewriter.create<CmpIOp>(loc, CmpIPredicate::ule, filterSize, c8);
+    Value cond16 =
+        rewriter.create<CmpIOp>(loc, CmpIPredicate::ule, filterSize, c16);
+    Value cond32 =
+        rewriter.create<CmpIOp>(loc, CmpIPredicate::ule, filterSize, c32);
+
```
Comment (zhanghb97) [inline midend/lib/Conversion/DAPVectorization/DAPVectorization.cpp:85]: // clang-format off

File: midend/lib/Conversion/DAPVectorization/DAPVectorization.cpp [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        });
+
+        builder.create<scf::YieldOp>(loc);
+    });
+
+    rewriter.eraseOp(op);
+    return success();
+  }
+};
+
```
Comment (zhanghb97) [inline midend/lib/Conversion/DAPVectorization/DAPVectorization.cpp:155]: // clang-format on

File: midend/lib/Utils/DAPUtils.cpp [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+      [&](OpBuilder &builder, Location loc, Value iv, ValueRange iargs) {
+        Value in_elem = builder.create<memref::LoadOp>(loc, input, iv);
+        Value vecIn_move_right = builder.create<vector::ShuffleOp>(
+            loc, iargs[0], iargs[0], arrayRef);
+        Value vecIn_next = builder.create<vector::InsertElementOp>(
+            loc, in_elem, vecIn_move_right, c0);
+        Value vecOut_next =
+            builder.create<vector::FMAOp>(loc, vecB0, vecIn_next, iargs[1]);
+
+        Value vecS1_lhs =
```
Comment (zhanghb97) [inline midend/lib/Utils/DAPUtils.cpp:120]: lhsVecS1

File: midend/lib/Utils/DAPUtils.cpp
Comment (zhanghb97) [inline midend/lib/Utils/DAPUtils.cpp:1]: clang-format

**Review by zhanghb97 — COMMENTED** [review]
- Modify variable name

**Pull request merged**